### PR TITLE
Refactor Activation Function implemented in ann/layers.

### PR DESCRIPTION
**Pull request by kartikdutt18**
Hi everyone,
In reference to [this discussion](https://github.com/mlpack/mlpack/pull/2174/files#r384141238), I have restructured code for activation functions implemented as layers as suggested by @zoq.
Kindly have a look.
Thanks.

### CODE REVIEW

File: src/mlpack/methods/ann/layer/elu_impl.hpp [hunk 1]
```diff
@@ -51,12 +51,27 @@ template<typename InputType, typename OutputType>
 void ELU<InputDataType, OutputDataType>::Forward(
     const InputType&& input, OutputType&& output)
 {
-  Fn(input, output);
-
-  if (!deterministic)
+  output.set_size(arma::size(input));
+  for (size_t i = 0; i < input.n_elem; i++)
```
Comment (prince776) [inline src/mlpack/methods/ann/layer/elu_impl.hpp]: I think we should avoid use of for loops and use armadillo methods as using GPU can be faster. I can be wrong.

Comment (kartikdutt18) [reply src/mlpack/methods/ann/layer/elu_impl.hpp]: I think that should be fine, I have benchmarking on my to do list (already had a PR open for the same), Once the same is established I will make changes for all activation / loss functions. Currently this is just a simple restructuring to make the code a bit more consistent. Thanks.

Comment (kartikdutt18) [reply src/mlpack/methods/ann/layer/elu_impl.hpp]: Hmmm, Should I make the changes now?
Well, I don't think I can resist taking a much closer look now. I will run some tests again and hopefully we can change everything to `arma` functions.

Comment (prince776) [reply src/mlpack/methods/ann/layer/elu_impl.hpp]: I would be happy to collaborate in this task. After all there are a lot of files to be restructured

Comment (zoq) [reply src/mlpack/methods/ann/layer/elu_impl.hpp]: > Hmmm, Should I make the changes now?

Don't think it should be part of the PR.

**Comment by zoq** [comment]
Looks good to me, is this PR ready from your side, or do you like to add anything else?

**Comment by saraansh1999** [comment]
I have'nt used armadillo before but would like to learn and help in the task of converting loops to arma funcitons. I am familiar with the coding style(I have done a lot of numpy in python), jut not the exact syntax of arma.
Is it possible for me to be involved in this?

File: src/mlpack/methods/ann/layer/elu_impl.hpp (hunk 1 above)
Comment (kartikdutt18) [reply src/mlpack/methods/ann/layer/elu_impl.hpp]: Agreed, So I'll keep it as it is for now, will reopen my previous PR after getting fixed benchmarks.
@prince776, I'll tag you in the PR once I reopen it and we can work on it.

**Comment by kartikdutt18** [comment]
> Looks good to me, is this PR ready from your side, or do you like to add anything else?

Great, I think this works for now, After proper benchmarking we can make changes in a different PR.
Thanks for the review.

**Comment by kartikdutt18** [comment]
> I have'nt used armadillo before but would like to learn and help in the task of converting loops to arma funcitons. I am familiar with the coding style(I have done a lot of numpy in python), jut not the exact syntax of arma.
> Is it possible for me to be involved in this?

@saraansh1999, Sure you can work on the issue, I'm in my hostel right now, once I get back home in the next couple of days, I'll redo the benchmarking and hopefully we can move ahead with the conversion. It will be part of a different PR though so I'll tag you in the same. Thanks.

**Comment by prince776** [comment]
> I have'nt used armadillo before but would like to learn and help in the task of converting loops to arma funcitons. I am familiar with the coding style(I have done a lot of numpy in python), jut not the exact syntax of arma.
> Is it possible for me to be involved in this?

@saraansh1999 I think it'll be a good idea to have a look at armadillo documentation, it's really helpful

File: src/mlpack/methods/ann/layer/elu_impl.hpp (hunk 1 above)
Comment (prince776) [reply src/mlpack/methods/ann/layer/elu_impl.hpp]: looking forward to it.

**Comment by zoq** [comment]
Thanks again for the contribution!

**Pull request merged**